File: src/cpp/3301.cpp
#include <iostream>

using namespace std;

int main() {
	int h, z, l;

	cin >> h >> z >> l;

	if (((h > z) and (h < l)) or ((h < z) and (h > l))) {
		cout << "huguinho" << endl;
	}

	if (((z > h) and (z < l)) or ((z > l) and (z < h))) {
		cout << "zezinho" << endl;
	}

	if (((l > h) and (l < z)) or ((l > z) and (l < h))) {
		cout << "luisinho" << endl;
	}

	return 0;
}

File: src/java/3301.java
import java.io.IOException;
import java.util.Scanner;
 
public class Main {
    public static void main(String[] args) throws IOException {
        int h, z, l;
        Scanner input = new Scanner(System.in);
        
        h = input.nextInt();
        z = input.nextInt();
        l = input.nextInt();
        
        if ((h > z && h < l) || (h < z && h > l)) {
            System.out.println("huguinho");
        }
    
        if ((z > h && z < l) || (z < h && z > l)) {
            System.out.println("zezinho");
        }
        
        if ((l > h && l < z) || (l < h && l > z)) {
            System.out.println("luisinho");
        }
    }
}
